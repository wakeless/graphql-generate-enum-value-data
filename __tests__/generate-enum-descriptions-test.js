const generateScalars = require('../src/generateEnumValueData');
const parseSchema = require('../src/parseSchema');
const fs = require('fs');
const path = require('path');

let types;

beforeEach(async () => {
  const rawSchema = fs.readFileSync(path.resolve('./__tests__/fixtures/schema.graphql'), 'utf8');
  schema = await parseSchema(rawSchema, './__tests__/fixtures/schema.graphql');
  types = schema.data.__schema.types;
});


describe('Generate assets', () => {
  it('should generate enum descriptions', () => {
    expect(generateScalars(types)).toMatchSnapshot();
  });
});
