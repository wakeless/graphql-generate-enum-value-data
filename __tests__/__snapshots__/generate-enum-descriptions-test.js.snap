// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generate assets should generate enum descriptions 1`] = `
"// @flow

/**
 * THIS FILE IS AUTO GENERATED, DO NOT EDIT DIRECTLY.
 */

export const BaseEnum = {
  customer_type: {
    name: 'customer_type',
    description: 'A cool customer type',
    isDeprecated: false,
    deprecationReason: null,
  },
  something_else: {
    name: 'something_else',
    description: null,
    isDeprecated: false,
    deprecationReason: null,
  },
};
"
`;
